test/unit/alignment/pairwise/alignment_result_test.cpp

```diff
@@ -268,3 +268,12 @@ TEST(alignment_result_test, empty_type)
     seqan3::alignment_result tmp(tr);
     // members must not be accessed
 }
+
+TEST(alignment_result_test, access_result_value_type)
+{
+    seqan3::detail::alignment_result_value_type result_value{2u, 5};
+    seqan3::alignment_result result(result_value);
+
+    using result_value_t = typename seqan3::detail::alignment_result_value_type_accessor<decltype(result)>::type;
```

Comment:
Sorry, one more question.
Isn't this equivalent without the need of the accessor struct:
```suggestion
    using result_value_t = seqan3::list_traits::front<seqan3::detail::transfer_template_args_onto_t<decltype(result)>, seqan3::type_list>;
```

Reply:
Sure, it does the same thing.
But how does this help the reader of the code to understand what happens? 
In addition, the combiler must instantiate first a type list to and then also instantiate on that a front type which is clearly more work and then it isn't documented.

Reply:
But why do we have those meta type traits if we shouldn't make use of them because they are more overhead?

Reply:
No, they are useful when they are useful. For example when I work with tuples or type lists. But now I don't work wtih tuples or type lists. I don't need a generic solution for this because the trait is bound to the alignment resutlt.